pkgs/dart_mcp/lib/src/api/tools.dart

```diff
@@ -197,6 +197,78 @@ enum JsonType {
   final String typeName;
 }
 
+/// Enum representing the types of validation failures when checking data
+/// against a schema.
+enum ValidationErrorType {
+  // General
+  typeMismatch,
+
+  // Schema combinators
+  allOfNotMet,
+  anyOfNotMet,
+  oneOfNotMet,
+  notConditionViolated,
+
+  // Object specific
+  requiredPropertyMissing,
+  additionalPropertyNotAllowed,
+  minPropertiesNotMet,
+  maxPropertiesExceeded,
+  propertyNamesInvalid,
+  propertyValueInvalid,
+  patternPropertyValueInvalid,
+  unevaluatedPropertyNotAllowed,
+
+  // Array/List specific
+  minItemsNotMet,
+  maxItemsExceeded,
+  uniqueItemsViolated,
+  itemInvalid,
+  prefixItemInvalid,
+  unevaluatedItemNotAllowed,
+
+  // String specific
+  minLengthNotMet,
+  maxLengthExceeded,
+  patternMismatch,
+
+  // Number/Integer specific
+  minimumNotMet,
+  maximumExceeded,
+  exclusiveMinimumNotMet,
+  exclusiveMaximumExceeded,
+  multipleOfInvalid,
+}
+
+/// A validation error with detailed information about the location of the
+/// error.
+extension type ValidationError.fromMap(Map<String, Object?> _value) {
+  factory ValidationError(
+    ValidationErrorType error, {
+    List<String>? path,
+    String? details,
+  }) => ValidationError.fromMap({
+    'error': error,
```

Comment:
This needs to be the serialized form of the error enum (either its name or index usually), otherwise this map isn't serializable